Content.Shared/_CM14/Scoping/ScopeComponent.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Robust.Shared.GameStates;
+using Robust.Shared.Map;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+
+namespace Content.Shared._CM14.Scoping;
+
+[RegisterComponent, NetworkedComponent, AutoGenerateComponentState]
+public sealed partial class ScopeComponent : Component
+{
+    /// <summary>
+    /// The entity that's scoping
+    /// </summary>
+    [ViewVariables, AutoNetworkedField]
+    public EntityUid? User;
+
+    /// <summary>
+    /// Is it currently scoped in
+    /// </summary>
+    [ViewVariables, AutoNetworkedField]
+    public bool IsScoping;
+
+    /// <summary>
+    /// Value to which zoom will be set when scoped in
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public float Zoom = 1f;
+
+    /// <summary>
+    /// Last position of the user when scope was toggled on
+    /// </summary>
+    [ViewVariables, AutoNetworkedField]
+    public EntityCoordinates LastScopedAt;
+
+    /// <summary>
+    /// Helper entity, which is spawned to load far chunks
+    /// </summary>
+    [ViewVariables, AutoNetworkedField]
+    public EntityUid? PvsLoader;
+
+    [DataField(customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>))]
+    public string ScopingToggleAction = "CMActionToggleScope";
```

Comment:
This can be made of type EntProtoId, then you don't need to specify the custom type serializer and other code can have more information about what this field is supposed to represent. One example would be being able to index it in IPrototypeManager without having to remember to specify EntityPrototype directly.
```suggestion
    [DataField]
    public EntProtoId ScopingToggleAction = "CMActionToggleScope";
```

Reply:
Much better, thanks!